apps/src/sites/code.org/pages/public/learn/local.js

```diff
@@ -196,80 +204,98 @@ function loadMap(locations) {
-      var controlList = document.createElement('ul');
-      controlList.classList.add('ullist');
-      controlList.classList.add('controls');
-      controlList.style = 'margin: 0px; padding: 0px; list-style-type: none;';
-      var viewAll = document.createElement('li');
-      viewAll.classList.add('active');
-      var viewAllLink = document.createElement('a');
-      viewAllLink.setAttribute('data-load', 'all');
-      viewAllLink.id = 'ullist_a_all';
-      viewAllLink.style = linkStyle;
-      viewAllLink.addEventListener('click', function(e) {
-        var activeItem = document.querySelector('li.active');
-        if (activeItem) {
-          activeItem.classList.remove('active');
-        }
-        viewAll.classList.add('active');
-        clearPopUp();
-        gmap.flyTo({
-          center: [lng, lat],
-          zoom: 10
-        });
-      });
-      var titleSpan = document.createElement('span');
-      titleSpan.innerHTML = 'View All';
-      viewAllLink.append(titleSpan);
-      viewAll.append(viewAllLink);
-      controlList.append(viewAll);
+      const linkStyle = {
+        color: 'rgb(102, 102, 102)',
+        display: 'block',
+        padding: '5px',
+        fontSize: 'inherit',
+        textDecoration: 'none',
+        cursor: 'pointer'
+      };
+      var allFeatures = [];
       featureList.forEach((feature, index) => {
-        var listItem = document.createElement('li');
-        listItem.id = `control-${index}`;
-        var link = document.createElement('a');
-        link.style = linkStyle;
-        var titleSpan = document.createElement('span');
-        titleSpan.innerHTML = feature.properties.title;
-        link.append(titleSpan);
-        listItem.append(link);
-        controlList.append(listItem);
-        link.addEventListener('click', function(e) {
-          flyToStore(feature);
-          createPopUp(feature);
-
-          var activeItem = document.querySelector('li.active');
-          if (activeItem) {
-            activeItem.classList.remove('active');
-          }
-          listItem.classList.add('active');
-        });
+        allFeatures.push(
+          <li id={`control-${index}`} key={`control-${index}`}>
+            <a
+              style={linkStyle}
+              onClick={event => {
+                flyToStore(feature);
+                createPopUp(feature);
+
+                var activeItem = document.querySelector('li.active');
+                if (activeItem) {
+                  activeItem.classList.remove('active');
+                }
+                $(event.target)
+                  .closest('li')
+                  .addClass('active');
+              }}
+            >
+              <span
+                // necessary to translate the character references in the title to the actual characters
+                // eslint-disable-next-line react/no-danger
+                dangerouslySetInnerHTML={{__html: feature.properties.title}}
```

Comment:
I believe these are user inputs we're displaying, no? I'm definitely concerned about using `dangerouslySetInnerHTML` here. I don't fully understand the comment as to why we need this, can you give an example?

Reply:
Sure! When I set the span text to Caesar's Palace, I get the character reference for ' instead of the actual character
<img width="1440" alt="Screen Shot 2020-10-06 at 1 11 06 PM" src="https://user-images.githubusercontent.com/17147070/95254868-6773c180-07d5-11eb-9a87-cd52d423e84e.png">

Reply:
Moved it into a React component and used SafeMarkdown like you recommended!